spopt/locate/p_median.py

```diff
@@ -533,3 +535,576 @@ def solve(self, solver: pulp.LpSolver, results: bool = True):
+        weights_cols: str,
+        p_facilities: int,
+        facility_capacity_col: str = None,
+        distance_metric: str = "euclidean",
+        name: str = "k-nearest-p-median",
+    ):
+        """
+        Set class variables for KNearestPMedian using input data.
+
+        Parameters
+        ----------
+        k_list : np.array
+            An array of integers representing the list of k values for each client.
+        gdf_demand : GeoDataFrame
+            A GeoDataFrame containing demand points with their associated attributes.
+        gdf_fac : GeoDataFrame
+            A GeoDataFrame containing facility points with their associated attributes.
+        demand_col : str
+            The column name in gdf_demand representing the coordinate of each client.
+        facility_col : str
+            The column name in gdf_fac representing the coordinate of each facility.
+        weights_cols : str
+            The column name in gdf_demand representing the weights for each client.
+        p_facilities: int
+           The number of facilities to be located.
+        facility_capacity_col : str, optional
+            The column name in gdf_fac representing the capacity of each facility,
+            by default None.
+        distance_metric : str, optional
+            The distance metric to be used in calculating distances between clients
+            and facilities, by default "euclidean".
+        name : str, optional
+            The name of the problem, by default "k-nearest-p-median".
+
+        Returns
+        -------
+        KNearestPMedian
+            The KNearestPMedian class itself with new class variables.
+
+        Warnings
+        --------
+        - The GeoDataFrames must have a valid CRS to perform distance calculations
+        accurately. And the CRS of gdf_demand and gdf_fac must be the same one.
+
+        Examples
+        --------
+        >>> from spopt.locate import KNearestPMedian
+        >>> import geopandas
+
+        Create the input data and attributes.
+
+        >>> k_list = np.array([1, 1])
+        >>> demand_data = {
+        ...    'ID': [1, 2],
+        ...    'geometry': [Point(0.5, 1), Point(1.5, 1)],
+        ...    'demand': [1, 1]}
+        >>> facility_data = {
+        ...    'ID': [101, 102],
+        ...    'geometry': [Point(1,1), Point(0, 2), Point(2, 0)],
+        ...    'capacity': [1, 1, 1]}
+        >>> gdf_demand = geopandas.GeoDataFrame(demand_data, crs='EPSG:4326')
+        >>> gdf_fac = geopandas.GeoDataFrame(facility_data, crs='EPSG:4326')
+
+        Create and solve a ``KNearestPMedian`` instance from the geodataframe.
+
+        >>> k_nearest_pmedian = KNearestPMedian.from_geodataframe(
+        ...     k_list, gdf_demand, gdf_fac,'geometry','geometry',
+        ...     demand_col='ID', facility_col='ID', weights_cols='demand',
+        ...     2, facility_capacity_col='capacity')
+        >>> k_nearest_pmedian = k_nearest_pmedian.solve(pulp.PULP_CBC_CMD(msg=False)
```

Comment:
Just a minor typo. It's missing the parenthesis in the last column. `k_nearest_pmedian.solve(pulp.PULP_CBC_CMD(msg=False))`.